sky/skylet/providers/gcp/config.py

```diff
@@ -814,11 +829,30 @@ def get_usable_vpc(config) -> str:
             )
             # Should not reach here.
 
+    subnets_all = _list_subnets(config, compute)
+
+    # Check if VPC for subnet has sufficient firewall rules.
     usable_vpc_name = None
-    for vpc in vpcnets_all:
-        if _check_firewall_rules(vpc["name"], config, compute):
-            usable_vpc_name = vpc["name"]
-            break
+    usable_subnet = None
+    sufficient_subnets = []
+    vpcnet_has_sufficient_rules = {}
+    for subnet in subnets_all:
+        vpc_name = subnet["network"].split("/")[-1]
```

Comment:
nit: Probably worth writing
```
subnet_to_vpc_name = lambda s: s["network"].split("/")[-1]
```
and replace L840, L853?

Reply:
Modified. Thanks!